Channel shuffle v4

lightningterror opened the pull request.

tadanokojin commented:
render doc files 
https://drive.google.com/file/d/19pwbolrzK1v3LsTDj_KOxc5FW8CchX5i/view?usp=sharing

KrossX committed:
tfx.fx: Port channel shuffle shaders from OpenGL.

Port RGBA shaders to support channel shuffle on Direct3D11.

lightningterror committed:
GSdx-d3d: Add the channel bit to support channel shuffle.

lightningterror committed:
GSdx-d3d11: Add the shader fetch in GSTextureFX11.

Allows us to fetch the channel from the tfx shader.

lightningterror committed:
GSdx-d3d: Add support for ChannelShuffle in PS constant buffer.

Needed for GXBY channel.

lightningterror committed:
GSdx-d3d: Add support for channel shuffle in GSRendererDX.

Split the code from GSRendererDX to GSRendererDX9 and GSRendererDX11. We
ensure d3d9 doesn't blow up with regressions, add required code to
GSRendererDX11 to properly support channel shuffle.

Note the feature is still not yet complete, copy function needs to be
implemented (suggested by Gregory) but it can be done at a later date,
this still fixes a bunch of issues on various games.

lightningterror committed:
tfx.fx: Add shaders support for PS_DEPTH_FMT on channel shuffle.

Red, blue and gXbY channels support it.

lightningterror committed:
GSdx-d3d11: Add support for depth_fmt bit as well as macro for shader.

lightningterror committed:
GSdx-d3d11: Add support for depth convert format in texture sampler.

Idea is to allot partial support of depth fmt on the new depth convert
format, this way we can utilize channel shuffle on Red, Blue or gXbY
channels with depth shader conversion.

Terminator 3 hits this code path and seems to fix a transparent wall.
Other games will probably benefit from this as well.

lightningterror committed:
GSdx-d3d: Add fixme note fo rPSSetShaderResource slot 4 channel shuffle.
